gcp_variant_transforms/transforms/extract_variant_header.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2017 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A PTransform to output a PCollection of single ``HeaderFields`` extracted
+from ``Variant`` recoreds. This ``HeaderFields`` contains only fields (FORMAT,
+INFO) not already defined in the header section of the VCF files."""
+
+from __future__ import absolute_import
+
+import apache_beam as beam
+
+from vcf.parser import _Format as Format
+from vcf.parser import _Info as Info
+from vcf.parser import field_counts
+
+from gcp_variant_transforms.beam_io import vcf_header_io
+from gcp_variant_transforms.transforms import merge_headers
+
+
+__all__ = ['ExtractHeaderFieldsFromVariant']
+
+
+_FLAG = 'Flag'
+_STRING = 'String'
+_NO_DESCRIPTION = 'no_description'
+_UNKNOWN_SOURCE = 'unknown_source'
+_UNKNOWN_VERSION = 'unknown_version'
+
+
+class _ExtractHeaderFieldsFromVariant(beam.DoFn):
+  """Extracts header fields from ``Variant`` record."""
+
+  def _is_defined_info_field(self, field_key, headers):
+    return headers and field_key in headers.infos
+
+
+  def _is_defined_format_field(self, field_key, headers):
+    return headers and field_key in headers.formats
```

Comment:
In what situations can `headers` be None? Even if it can be, I think it's better to just create an empty VcfHeader and avoid checking for None here.

Reply:
If no headers is passed? In our pipeline this won't happen because we pass it as side input (which has default value). This was to be defensive, we can remove it if you don't like it.